src/dmd/expressionsem.d

```diff
@@ -5168,7 +5168,10 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
         {
             if (!sc.insert(s))
             {
-                e.error("declaration `%s` is already defined", s.toPrettyChars());
+                auto conflict = sc.search(Loc.initial, s.ident, null);
+                e.error("declaration `%s` is already defined as `%s` `%s` in `%s`:`%d`",
```

Comment:
Can the previous definition be in another module ? I ask because I don't like that fact to have twice the same filname. If the answer is no then a lighter message would be welcome, e.g

    "declaration `%s` is already defined as `%s`  at line `%d`",

Reply:
Maybe split the error message into two lines instead? (It's rather long anyways)

E.g.
```
fail_compilation/already_defined.d(4): Error: declaration `already_defined.func1.a` is already defined as `variable` `a` in `fail_compilation/already_defined.d`:`3`
```
becomes
```
fail_compilation/already_defined.d(4): Error: declaration `already_defined.func1.a` is already defined
fail_compilation/already_defined.d(3):        `variable` `a` is declared here
```

Reply:
Both good points, I'll just go with @MoonlightSentinel 's